### ISSUE TITLE
Missing null-check on Equals causes issues when used with EF

### ISSUE THREAD

**Issue by devpikachu** [medo64/Medo.Uuid7#1]
Missing null-check on `Equals` method causes issues when used with EF, as EF performs an `.Equals` call with a null value when adding a new entity to know if it has to generate the key.

**Comment by medo64** [comment]
Can you please clarify which issue this causes on EF?

Reason I ask is because I see no need for a null check. The first if checks if object is `Uuid7` (if true, obj is not null). Second if checks if object is `Guid`. Anything else, including null, simply returns false. Having null check at the beggining would not do anything different.

```
    public override bool Equals([NotNullWhen(true)] object? obj) {
        if (obj is Uuid7 uuid) {
            return CompareArrays(Bytes, uuid.Bytes) == 0;
        } else if (obj is Guid guid) {
            return CompareArrays(Bytes, guid.ToByteArray()) == 0;
        }
        return false;
    }
```

**Comment by devpikachu** [comment]
Of course! It seems I was mistaken as to where this error is happening. It also seems that EF Core is doing some weird stuff to verify if a property has a default value.

The call stack (after propagating through various EF Core internals) ends up here:
https://github.com/medo64/Medo.Uuid7/blob/ad38a971a9c2dc289880321c41b9d3d0d5a2ad6f/src/Uuid7.cs#L492

Interestingly, if you breakpoint here, the `other` param simply throws a NullRef exception. This is what I meant by EF Core doing something odd.

This then propagates to the `CompareArrays` method, which throws here:
https://github.com/medo64/Medo.Uuid7/blob/ad38a971a9c2dc289880321c41b9d3d0d5a2ad6f/src/Uuid7.cs#L552

This is due to accessing the `Length` property of `buffer2` which is now null.

The way I, perhaps hackily, fixed this locally is by adding the following check to the `CompareArrays` method:

```csharp
if (buffer1 is null || buffer2 is null)
{
    return -1;
}
```

I would assume null-checking `buffer1` is completely redundant, but I haven't verified this at this time. If you'd like, I can verify this locally as I already have EF Core setup with Uuid7 as PKs.

**Comment by medo64** [comment]
Thanks for reporting this, Will fix after I have my morning coffee. :)

**Comment by medo64** [comment]
Should be fixed in commit https://github.com/medo64/Medo.Uuid7/commit/ae0e9a0e2268e105c4ead6cd7d998270843fd2a4.
Thanks for reporting it!

**Comment by medo64** [comment]
New release should also be on Nuget now.

**Comment by devpikachu** [comment]
Confirmed to be fixed with version 1.3.0 of the NuGet Package. Thank you so much for looking into this!
